sdk/core/core-tracing/review/core-tracing.api.md

```diff
@@ -22,7 +22,8 @@ export interface ContextAPI {
 // @public
 export function createSpanFunction(args: CreateSpanFunctionArgs): <T extends {
     tracingOptions?: OperationTracingOptions | undefined;
-}>(operationName: string, operationOptions: T | undefined) => {
+    spanOptions?: SpanOptions | undefined;
+}>(operationName: string, operationOptions?: T | undefined) => {
```

Comment:
Usage example (re-adding because my comment got stale):


```ts
// without customization
const span = createSpan("deleteConfigurationSetting", operationOptions);
// setting spanKind to SERVER on the newly created span
const span = createSpan("deleteConfigurationSetting", {...operationOptions, spanOptions: { spanKind: SpanKind.SERVER } });

// adding custom attributes / links:
const spanOptions = {
    attributes: { "foo": "bar" },
    kind: SpanKind.CLIENT,
    links: [link1, link2]
}
const span = createSpan("deleteConfigurationSetting", {...operationOptions, ...spanOptions })
```

Alternatives I considered:

```ts
// separate param to `createSpan`
const span = createSpan("deleteConfigurationSetting", operationOptions, { spanKind: SpanKind.CLIENT });

// nested spanOptions inside tracingOptions:
const createSpanOptions = {
    ...operationOptions
    tracingOptions: {
        ...operationOptions.tracingOptions,
        spanOptions: { kind: SpanKind.SERVER }
    }
}
```

Reply:
I like the separate param version personally:

```ts
// separate param to `createSpan`
const span = createSpan("deleteConfigurationSetting", operationOptions, { spanKind: SpanKind.CLIENT });
```

This makes the most sense to me, since smashing additional configuration onto `OperationOptions` doesn't buy us much, and it's definitely optional